### PR TITLE
Infinite scroll

### PR DESCRIPTION
**Pull request by pitiphong-p**
This PR introduces the new data structure that will handle the `Infinite Scroll` or `Load More` logic

### CODE REVIEW

File: OmiseSwift/List.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
+ import Foundation
```
Comment (chakrit) [inline OmiseSwift/List.swift]: whitespace ☝️

File: OmiseSwift/List.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    var loadedIndices = 0..<0
+    public let order: Ordering
+    
+    public var limit: Int = 0
+    public var total: Int = 0
+    public var data: [TItem] = [] {
+        didSet {
+            dataUpdatedHandler?(data)
+        }
+    }
```
Comment (chakrit) [inline OmiseSwift/List.swift:16]: Can some of these properties be made `private set` or immutable?

File: OmiseSwift/Listable.swift [hunk 3]
```diff
@@ -50,4 +52,105 @@ public extension Listable where Self: ResourceObject {
...
+            case .success(let result):
+                let list = List(endpoint: operation.endpoint, paths: operation.pathComponents, order: listParams?.order, list: result)
+                callback(.success(list))
+            case .fail(let error):
+                callback(.fail(error))
+            }
+        }
+        
+        return client.call(operation, callback: requestCallback)
+    }
```
Comment (chakrit) [inline OmiseSwift/Listable.swift:73]: ☝️ Can this use `Failable.map`?

**Review by chakrit — DISMISSED** [review]

**Comment by chakrit** [comment]
ตาลาย 😵

File: OmiseSwift/Listable.swift (hunk 3 above)
Comment (pitiphong-p) [reply OmiseSwift/Listable.swift:73]: I think we can't use that. This closure will call the given callback not to convert it though

File: OmiseSwift/List.swift (hunk 2 above)
Comment (pitiphong-p) [reply OmiseSwift/List.swift:16]: That's a good catch. Thank you
BTW I already fixed it

File: OmiseSwift/Listable.swift (hunk 3 above)
Comment (chakrit) [reply OmiseSwift/Listable.swift:73]: :trollface:

File: OmiseSwift/Listable.swift [hunk 4]
```diff
@@ -50,4 +52,105 @@ public extension Listable where Self: ResourceObject {
...
+extension List where TItem: ResourceObject, TItem: Listable {
+    public func loadNextPage(using given: Client? = nil, count: Int? = nil, callback: @escaping (Failable<[TItem]>) -> Void) {
+        TItem.loadNextPage(list: self, using: given, count: count) { (result) in
+            switch result {
+            case .success(let addedValues):
+                callback(.success(addedValues))
+            case .fail(let error):
+                callback(.fail(error))
+            }
+        }
```
Comment (chakrit) [inline OmiseSwift/Listable.swift:134]: can `.map` here and (the loadPrevious method) too I think ☝️

**Review by chakrit — CHANGES_REQUESTED** [review]

**Comment by pitiphong-p** [comment]
Thank you and fixed 🙇

File: OmiseSwift/Listable.swift (hunk 3 above)
Comment (pitiphong-p) [reply OmiseSwift/Listable.swift:73]: Fixed :)

File: OmiseSwift/Listable.swift (hunk 4 above)
Comment (pitiphong-p) [reply OmiseSwift/Listable.swift:134]: Fixed :)

**Review by chakrit — APPROVED** [review]

**Review by igroomgrim — APPROVED** [review]
👍  🔮

**Pull request merged**